### PR TITLE
Conslidate normative lanugage on x5t and x5chain usage (plus SecConSec content)

### LINKED ISSUE #222: Clarify if/when X5Chain should be included
At the IETF 119 Hackathon, it became confusing when/if `33 x5chain` should be added to the protected header, the unprotected header, or not at all.

While adding the x5chain to the statement can help discoverability, it expands the CBOR content, which is intended to be as small as possible.
Placing x5Chain in the protected header also means it's included in the signature, which really adds no value.
Alternatives are to place it in the unprotected header, which isn't signed and could be stripped, but typically isn't in practice, so it still adds bloat.

Lastly, the spec speaks to different options: `x5t` (34), `x5chain`(33) or `kid`(4) 
[Line: 404](https://github.com/ietf-wg-scitt/draft-ietf-scitt-architecture/blob/main/draft-ietf-scitt-architecture.md?plain=1#L404) says:
> During registration, a Transparency Service MUST, at a minimum, authenticate the Issuer of the Signed Statement by validating the CBOR signature and checking the identity of the issuer against one of its currently configured trust anchors, using the `x5t` (34), `x5chain`(33) or `kid`(4) protected headers of the Signed Statement as hints.
> For instance, in order to authenticate X.509 Signed Statements, the Transparency Service MUST build and validate a complete certificate chain from the Issuer's certificate identified by `x5t`, to one of the root certificates most recently registered as a trust anchor of the Transparency Service.

Proposal would be to:
1. Move x5Chain to the unprotected header
1. Remove x5Chain
1. Update the samples to show the different combinations. The [current sample](https://github.com/ietf-wg-scitt/draft-ietf-scitt-architecture/blob/main/draft-ietf-scitt-architecture.md?plain=1#L523) shows all as options, which has confused implementers.

### PR DESCRIPTION
**Pull request by henkbirkholz**
fixes #222

### CODE REVIEW

File: draft-ietf-scitt-architecture.md [hunk 1]
```diff
@@ -766,6 +770,10 @@ An example of Signed Statement's payloads that can improve authenticity assuranc
...
 
+Auditors should be aware that the certification path information included in an unprotected `x5chain` header of a to-be-registered Signed Statement can be tampered with by a malicious Transparency Service (e.g., one that does not incorporate remote attestation), which may replace the intermediate certificates and ultimately connect to an unexpected root.
+This modification can allow malicious TS to forge Claims that look genuine except for the wrong trust anchor.
+In addition to performing chain validation in accordance with PKIX rules specified in {{-PKIX}}, auditors should explicitly verify the root certificates are genuine.
```
Comment (aj-stein-nist) [inline draft-ietf-scitt-architecture.md:775]: Genuine sounds subjective.

```suggestion
In addition to performing chain validation in accordance with PKIX rules specified in {{-PKIX}}, auditors should explicitly verify the root certificates.
```

File: draft-ietf-scitt-architecture.md [hunk 2]
```diff
@@ -182,9 +184,9 @@ To ensure readability, only a core set of terms is included in this section.
 
 **Editor's Note:**: *The label "394" is expected to be reserved by this document, in the COSE Header Parameters Registry.*
 
-The terms "header", "payload", and "to-be-signed bytes" are defined in {{RFC9052}}.
+The terms "header", "payload", and "to-be-signed bytes" are defined in {{-COSE}}.
 
-The terms claim is defined in {{RFC8392}}, and is repeated here for readability:
+The term "claim" is defined in {{RFC8392}} and is repeated here for readability:
```
Comment (aj-stein-nist) [inline draft-ietf-scitt-architecture.md:189]: We also still have the term "Claim" in [the terminology section](https://ietf-wg-scitt.github.io/draft-ietf-scitt-architecture/x5chain-usage/draft-ietf-scitt-architecture.html#section-2). Should we not remove that and replace it with this wording? Because of his git diffs work I cannot recommend that here, or I would do it for you, but this seems like an important change to not cause confusion down the road.

**Review by aj-stein-nist — DISMISSED** [review]

File: draft-ietf-scitt-architecture.md (hunk 1 above)
Comment (henkbirkholz) [reply draft-ietf-scitt-architecture.md:775]: Not much lost here. I'd be okay with that suggestion.

File: draft-ietf-scitt-architecture.md [hunk 3]
```diff
@@ -471,9 +474,9 @@ For a software supply chain, payloads describing the software artifacts may incl
 Once all the Envelope headers are set, an Issuer MUST use a standard COSE implementation to produce an appropriately serialized Signed Statement.
 The SCITT tag `COSE_Sign1_Tagged` is outside the scope of COSE, and used to indicate that a signed object is a Signed Statement.
 
-Issuers may produce Signed Statements about different Artifacts under the same Identity.
+Issuers MAY produce Signed Statements about different Artifacts under the same Identity.
```
Comment (OR13) [inline draft-ietf-scitt-architecture.md:477]: This is a throwaway, consider just removing this normative statement.

**Review by OR13 — APPROVED** [review]

File: draft-ietf-scitt-architecture.md (hunk 3 above)
Comment (SteveLasker) [reply draft-ietf-scitt-architecture.md:477]: Keeping this line puts context to how multiple Signed Signed Statements can be used. 
Changing to a colocial "can" to keep the description, without making it vague normative language. 
```suggestion
Issuers can produce Signed Statements about different Artifacts under the same Identity.
```

**Comment by henkbirkholz** [comment]
good enough!

File: draft-ietf-scitt-architecture.md [hunk 4]
```diff
@@ -766,6 +770,10 @@ An example of Signed Statement's payloads that can improve authenticity assuranc
...
 
+Auditors should be aware that the certification path information included in an unprotected `x5chain` header of a to-be-registered Signed Statement can be tampered with by a malicious Transparency Service (e.g., one that does not incorporate remote attestation), which may replace the intermediate certificates and ultimately connect to an unexpected root.
+This modification can allow malicious TS to forge Claims that look genuine except for the wrong trust anchor.
+In addition to performing chain validation in accordance with PKIX rules specified in {{-PKIX}}, auditors should explicitly verify the root certificates.
```
Comment (CBonnell) [inline draft-ietf-scitt-architecture.md:775]:
```suggestion
Auditors MUST perform certification path validation in accordance with PKIX rules specified in {{-PKIX}}, In particular, auditors MUST verify that certification paths chain to a trusted root certificate.
```

File: draft-ietf-scitt-architecture.md [hunk 5]
```diff
@@ -766,6 +770,11 @@ An example of Signed Statement's payloads that can improve authenticity assuranc
...
 
+Auditors should be aware that the certification path information included in an unprotected `x5chain` header of a to-be-registered Signed Statement can be tampered with by a malicious Transparency Service (e.g., one that does not incorporate remote attestation), which may replace the intermediate certificates and ultimately connect to an unexpected root.
+This modification can allow malicious TS to forge Claims that look genuine except for the wrong trust anchor.
+Auditors MUST perform certification path validation in accordance with PKIX rules specified in {{-PKIX}}.
+In particular, auditors MUST verify that certification paths chain to one or more trust anchors (often represented as root certificates).
```
Comment (SteveLasker) [inline draft-ietf-scitt-architecture.md:776]: Editors call: agreeing to the content, question if this is the correct section, which we can move in another PR.

File: draft-ietf-scitt-architecture.md (hunk 2 above)
Comment (SteveLasker) [reply draft-ietf-scitt-architecture.md:189]: Agreed to remove claims in the terminology section.

**Review by SteveLasker — APPROVED** [review]
LGTM

**Review by CBonnell — APPROVED** [review]

**Review by aj-stein-nist — APPROVED** [review]
Thanks for the update. With the last commits and merge, I am happy with this change and have no other objections.
